numpy/tests/typing/pass/dtype.py

```diff
@@ -1,3 +1,33 @@
 import numpy as np
 
 np.dtype(dtype=np.int64)
+np.dtype(int)
+np.dtype("int")
+np.dtype(None)
+
+np.dtype((int, 2))
+np.dtype((int, (1,)))
+
+np.dtype({"names": ["a", "b"], "formats": [int, float]})
+np.dtype({"names": ["a"], "formats": [int], "titles": [object]})
+np.dtype({"names": ["a"], "formats": [int], "titles": [object()]})
+np.dtype(
+    {
+        "names": ["a", "b"],
+        "formats": [int, float],
+        "itemsize": 9,
+        "aligned": False,
+        "titles": ["x", "y"],
+        "offsets": [0, 1],
+    }
+)
+
+np.dtype({"field1": (float, 1), "field2": (int, 3)})
+np.dtype((np.float_, float))
+
+
+class Test:
+    dtype = float
+
+
+np.dtype(Test())
```

Comment:
maybe worth adding a list example here like : np.dtype([('name', np.unicode_, 16), ('grades', np.float64, (2,))]).

Didnt find it in the tests.

Reply:
Good call! Implemented in https://github.com/numpy/numpy/pull/16622/commits/12333c72aee600ba66387f7963dd73b4801e0fc0.